src/services/person.py

```diff
@@ -0,0 +1,51 @@
+import logging
+from functools import cache
+from typing import Optional, List
+
+from aioredis import Redis
+from elasticsearch import AsyncElasticsearch
+from elasticsearch_dsl import Search
+from fastapi import Depends
+
+from core.config import CACHE_TTL
+from db.cache import ModelCache
+
+from db.elastic import get_elastic
+from db.redis import get_redis
+from models.person import Person
+from services.base import BaseESService
+
+logger = logging.getLogger(__name__)
+
+
+class PersonService(BaseESService):
+    model = Person
+    index = 'persons'
+
+    async def search(self, search_query: str = "",
+                     filter_genre: Optional[str] = None,
+                     sort: Optional[str] = None,
+                     page_number: int = 1,
+                     page_size: int = 50) -> List[Person]:
+
+        s = Search(using=self.elastic, index=self.index)
+        if search_query:
+            s = s.query('match', full_name=search_query)
+        if sort:
+            s = s.sort(sort)
+        start = (page_number - 1) * page_size
+        query = s[start: start + page_size].to_dict()
+        films = await self.cache.get_by_elastic_query(query)
+        if not films:
```

Comment:
Опечатка, надо исправить: Здесь должны быть не films, а persons, и на 2 строки ниже
```suggestion
        persons = await self.cache.get_by_elastic_query(query)
        if not persons:
```